yt_dlp/utils.py

```diff
@@ -473,24 +473,19 @@ def get_elements_text_and_html_by_attribute(attribute, value, html, escape_value
     attribute in the passed HTML document
     """
 
+    value_quote_optional = '' if re.match(r'''[\s"'`=<>]''', value) else r'?'
+
     value = re.escape(value) if escape_value else value
 
-    retlist = []
-    for m in re.finditer(r'''(?xs)
-        <(?P<tag>[a-zA-Z0-9:._-]+)
-         (?:\s+[a-zA-Z0-9_:.-]+(?:=\S*?|\s*=\s*(?:"[^"]*"|'[^']*')|))*?
-         \s+%(attribute)s(?:=%(value)s|\s*=\s*(?P<_q>['"]?)%(value)s(?P=_q))
-         (?:\s+[a-zA-Z0-9_:.-]+(?:=\S*?|\s*=\s*(?:"[^"]*"|'[^']*')|))*?
-        \s*>
-    ''' % {'attribute': re.escape(attribute), 'value': value}, html):
-        content, whole = get_element_text_and_html_by_tag(m.group('tag'), html[m.start():])
+    partial_element_re = \
+        r'''<(?P<tag>[a-zA-Z0-9:._-]+)(?:\s+[a-zA-Z0-9_:.-]+(?:=[^\s"'`=<>]+|=(?:"[^"]*"|'[^']*')|))*?\s+%(attribute)s=(?P<_q>['"]%(vqo)s)%(value)s(?P=_q)''' % \
+        {'attribute': re.escape(attribute), 'value': value, 'vqo': value_quote_optional}
```

Comment:
Would this not work?
```suggestion
    # Do not use re.verbose since value may contain unescaped spaces
    partial_element_re = (
        r'''<(?P<tag>[a-zA-Z0-9:._-]+)(?:\s(?:[^>"']|"[^"]*"|'[^']*')+)?\s%(attribute)s\s*=\s*(?P<_q>['"]%(vqo)s)(?:%(value)s)(?P=_q)'''
        % {'attribute': re.escape(attribute), 'value': value, 'vqo': value_quote_optional})
```
It is simpler than your regex, passes all tests and should not cause any catastrophic backtracking as far as I can tell (since the `|` groups are mutually exclusive and there is no non-greedy matching)

Reply:
Let me check then.

Reply:
Thanks, this seems to work great!